packages/nimble-components/src/table/models/virtualizer.ts

```diff
@@ -58,6 +61,7 @@ export class Virtualizer<TData extends TableRecord = TableRecord> {
                 // by the same margin so the column headers align with the corresponding rendered cells
                 const viewportBoundingWidth = borderBoxSize.inlineSize;
                 this.headerContainerMarginRight = viewportBoundingWidth - this.table.viewport.clientWidth;
+                this.horizontalScrollbarHeight = borderBoxSize.blockSize - this.table.viewport.clientHeight;
```

Comment:
Will this sufficiently catch other conditions besides the viewport changing size that could result in the scrollbar appearing? For example in the text area it ended up being a whack-a-mole of properties that could trigger a scrollbar to appear: https://github.com/ni/nimble/blob/52caaabfa9c595f1909d1bb374aae3d228c4d172/packages/nimble-components/src/text-area/index.ts#L69-L74

Related, maybe we could use [`scrollbar-gutter`](https://caniuse.com/mdn-css_properties_scrollbar-gutter) to always reserve the space and simplify? https://github.com/ni/nimble/issues/1271

Reply:
> Using scrollbar-gutter was one of the first paths I went down in my attempt to fix this issue, but from what I can tell, it doesn't apply to horizontal scrollbars.

Well dang, seems to be the case 🤦‍♂️: https://github.com/w3c/csswg-drafts/issues/5252

> I manually tested the following scenarios, and they all caused the value of the token to change appropriately:
> 
> - toggle the value of column-visible on a column
> - add/remove a column from the DOM
> - increase/decrease the minimum width of a column
> - resize browser window

Sold, thanks for looking into it.